
From: James Prestwood <prestwoj@gmail.com>
To: iwd@lists.linux.dev
Cc: James Prestwood <prestwoj@gmail.com>
Subject: [PATCH 4/5] scan: warn on zero frequency when adding to request
Date: Thu, 24 Oct 2024 09:46:34 -0700
Message-Id: <20241024164635.176631-4-prestwoj@gmail.com>
X-Mailer: git-send-email 2.34.1
In-Reply-To: <20241024164635.176631-1-prestwoj@gmail.com>
References: <20241024164635.176631-1-prestwoj@gmail.com>
Precedence: bulk
X-Mailing-List: iwd@lists.linux.dev
List-Id: <iwd.lists.linux.dev>
List-Subscribe: <mailto:iwd+subscribe@lists.linux.dev>
List-Unsubscribe: <mailto:iwd+unsubscribe@lists.linux.dev>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit

This shouldn't ever happen, but if it does we want to alert the user.
---
 src/scan.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/scan.c b/src/scan.c
index 2ffbef6d..c1ccc234 100644
--- a/src/scan.c
+++ b/src/scan.c
@@ -292,6 +292,9 @@ static void scan_freq_append(uint32_t freq, void *user_data)
 {
 	struct scan_freq_append_data *data = user_data;
 
+	if (L_WARN_ON(!freq))
+		return;
+
 	l_genl_msg_append_attr(data->msg, data->count++, 4, &freq);
 }
 
-- 
2.34.1


